### PR TITLE
Feature/file labels

### PR DESCRIPTION
**Pull request by mateuszkrasucki**
- Adds support for file `labels`. 
- Labels are stored as part of the object metadata (in key). On new file creation, file update and file deletion `files collection` file (with JSON content) is created/updated per `label. Those files are treated as any other files, they are versioned etc. 
- `Files collection` collection files are not created/updated in cloud instance when new files are `synced` via sync part of storage API. Instead `files collection` files itself are being synced as any other file. 
(*This behaviour follows assumption that only one local instance uploads files to given bucket, i.e. patient belongs at given time to one local instance.*).
- One file can have multiple labels and labels can change in versions so they need to be specified with update call as well. 
- When passed in header and in form-data multiple labels are formatted as pipe-delimited, e.g.: `vitalSign|basicPatientInfo`.
- Following labels are defined in swagger API spec: 
    - basicPatientInfo
    - vitalSign
    - diagnoses
    - filesCollection
- Related: #8

### CODE REVIEW

File: docs/api/storage.yml [hunk 1]
```diff
@@ -91,6 +91,15 @@ paths:
...
           type: string
 
+        - in: formData
+          name: labels
+          description: Optional labels
+          required: false
+          type: array
+          items:
+            type: string
+          collectionFormat: pipes
```
Comment (dominikznidar) [inline docs/api/storage.yml]: Could we change this to comma separated. Seems more common to me.

File: docs/api/storage.yml [hunk 2]
```diff
@@ -118,6 +127,9 @@ paths:
       produces:
         - application/octet-stream
         - application/json; charset=utf-8
+        - text/openEhrXml
+        - text/openEhrJson
```
Comment (dominikznidar) [inline docs/api/storage.yml]: why are these two type listed here?

File: storage/s3/s3.go [hunk 3]
```diff
@@ -217,6 +218,9 @@ func (s *s3storage) List(_ context.Context, bucketID, prefix string) ([]*models.
 			return nil, errors.Wrap(info.Err, "Failed to read object from a list")
 		}
 
+		s.logger.Error().Msgf("%s", info)
+		s.logger.Error().Msgf("%s", info.ContentType)
+
```
Comment (dominikznidar) [inline storage/s3/s3.go]: are those debug statements?

File: docs/api/storage.yml (hunk 2 above)
Comment (mateuszkrasucki) [reply docs/api/storage.yml]: Hm, I've added it here cause this endpoint will produce this content-types. Afaik go-swagger generated server will overwrite content-type with application/octet-stream if it's not set here. Of course default producers can be also changed in code but decided that it's cleanest to just add it here. Let me know what you think.

File: storage/s3/s3.go (hunk 3 above)
Comment (mateuszkrasucki) [reply storage/s3/s3.go]: removed

File: docs/api/storage.yml (hunk 1 above)
Comment (mateuszkrasucki) [reply docs/api/storage.yml]: done

File: docs/api/storage.yml (hunk 2 above)
Comment (dominikznidar) [reply docs/api/storage.yml]: ok

**Pull request merged**